functions/Test-DbaWMConnection.ps1

```diff
@@ -0,0 +1,428 @@
+    
+    .PARAMETER ComputerName
+        The computer to test against.
+    
+    .PARAMETER Credential
+        The credentials to use when running the test.
+        Bad credentials are automatically cached as non-working, a behavior that can be disabled by the 'Cache.Management.Disable.BadCredentialList' configuration.
+    
+    .PARAMETER Type
+        The connection protocol types to test.
+        By default, all types are tested.
+        
+        Available connection protocol types:
+        "CimRM", "CimDCOM", "Wmi", "PowerShellRemoting"
+    
+    .PARAMETER ClearCache
+        Clears all previously cached connection information and creates a new cache for the computer processed.
+    
+    .PARAMETER Force
+        Overrides safeguards and "do you really want to do this" kinds of issues.
+        - Will force a test to proceed on a known bad credential
+    
+    .PARAMETER OverrideCredential
+        By default, explicitly defined credential trump stored credentials.
+        Setting the specified credentials as override causes the stored credentials to be used for every connection, no matter the input.
+        This can be used without credentials, in which case the default windows authentication of the current user will be used.
+    
+        Note: Given its nature, Test-DbaWmConnection will ignore this setting.
+    
+    .PARAMETER SetCredential
+        Sets the specified credentials (or windows authentication, if none were) as the default credential for the connection.
+        In situations where a credential is known to not work for this target, the function will then fall back to this registered credential.
+        When used in combination with OverrideCredential, the thus registered credential will be prioritized before input instead.
+    
+    .PARAMETER Silent
+        Replaces user friendly yellow warnings with bloody red exceptions of doom!
+        Use this if you want the function to throw terminating errors you want to catch.
+    
+    .EXAMPLE
+        PS C:\> Test-DbaWMConnection -ComputerName sql2014
+    
+        Performs a full-spectrum connection test against the computer sql2014.
+        The results will be reported and registered, future calls from Get-DbaWmObject will thus recognize the results and optimize the query.
+    
+    .EXAMPLE
+        PS C:\> Test-DbaWMConnection -ComputerName sql2014 -Credential $null -DisableConnectionType PowerShellRemoting, Wmi -OverrideCredential -SetCredential -Type CimDCOM, CimRM
+    
+        This test will run a connectivity test against the computer sql2014
+        - It will use windows authentication of the current user
+        - It will only test Cim over DCOM and Cim over WinRM
+        - It will disable the use of PowerShellRemoting and WMi as a connectivity type
+        - It will configure all connections to this server to use the current windows credentials, instead of what ever may be specified by the user or another function.
```

Comment:
These functions will be public, that is, users can use it directly, either manually or in scripts. So can our own functions of course. Basically, it's an internal function that is so useful, that we decided to make it available to our users directly, for their own needs.
Of course, settings of the connection management only affect calls made through our own connection functions. When a user directly uses Get-WmiObject, then that will use wmi and only with the connection information specified by the user, as usual. We do not override the regular commands. The very next line points out that this affects Get-DbaWmObject only.

Do I need to be more explicit than that?